### DISCUSSION TITLE
MetadataDiscovery.processDiscoveredEntities should NOT return map()

### DISCUSSION THREAD

**Discussion by why28zhang** [mikro-orm/mikro-orm#5808] [General]
### Describe the bug

MetadataDiscovery.processDiscoveredEntities()  modifies the metadata items in discovered param, BUT when it returned, it uses

`
return discovered.map(meta => this.metadata.get(meta.className)
`

thus all the modifications is restored, this especialy effected the result of MikroORM.discoverEntity, because it used this function's result:

`
const metadata = this.discovery.processDiscoveredEntities(tmp)
`

### Reproduction

1. use orm.discoverEntity() to add an entity, for example with name 'test'
2. then when calling orm.em.findAll('test'), it will throw an exception with message:

`
meta.trackingProps is not iterable
`

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.2.9

### Node.js version

20.15.0

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
I am not following what you are trying to report here, an actual (and complete) reproduction would help. We have many tests that are calling `orm.discoverEntity()` and work just fine.
